examples/threathunter/threat_intelligence/threatintel.py

```diff
@@ -0,0 +1,240 @@
+import logging
+from feed_helper import FeedHelper
+from cabby import create_client
+from cbapi.psc.threathunter import CbThreatHunterAPI, Report
+from cbapi.errors import ApiError
+from cbapi.psc.threathunter.models import Feed
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+class ThreatIntel:
+    def __init__(self):
+        self.cb = CbThreatHunterAPI()
+
+    def push_to_cb(self, feed_id, results):
+        try:
+            feed = self.cb.select(Feed, feed_id)
+        except ApiError as e:
+            log.error(f"couldn't find CbTH feed {feed_id}: {e}")
+            return
+
+
+        report_list_to_send = []
+        report_dict_list_to_validate = []
+
+        #optional_report_attributes = ["link", "tags", "iocs", "iocs_v2", "visibility"]
+
+        for result in results:
+            try:
+                report_dict = { #document these values and what a result should contain
+                    "id": str(result.id),
+                    "timestamp": int(result.timestamp.timestamp()),
+                    "title": str(result.title),
+                    "description": str(result.description),
+                    "severity": int(result.severity),
+                    "iocs_v2": [ioc_v2.as_dict() for ioc_v2 in result.iocs_v2]
+                }
+
+
+                # for opt in optional_report_attributes:
+                #     if hasattr(result, opt):
+                #         if 'opt' == 'iocs_v2' or 'opt' == 'iocs':
+                #             report_dict[opt] = [ioc_v2.as_dict() for ioc_v2 in result.iocs_v2]
+                #         else:
+                #             report_dict[opt] = getattr(result, opt)
+
+                report_dict_list_to_validate.append(report_dict)
+
+                report = Report(self.cb, initial_data=report_dict, feed_id=feed_id)
+                report_list_to_send.append(report)
+
+            except Exception as e:
+                log.error(f"Failed to create a report dictionary from result object. {e}")
+
+        validation = self.input_validation(report_dict_list_to_validate)
```

Comment:
By not performing the validation in the for loop above we add a O(2n) complexity since we have to go through the reports again

Reply:
I'm not sure if moving validation into the creation for loop above would reduce complexity. Either way, we have to send the report to the validation function before sending. Any other idea about how to reduce the complexity?

Reply:
How large is the list of reports? unless N is ginormous the difference between any coefficient on N is not worth the code changes to improve it. It's a diminishing return to improve that.

Reply:
After some further investigation, I am noticing that all reports will not be sent if a singular report fails. Is this the intended design? Do we want to have better support for identifying which reports failed so they can be fixed? How does this tie into the fetch loop will those reports be skipped over and the script will need to be executed again from the previous start time?

Reply:
I will adjust threatintel.py to validate reports individually, and also to log the report dictionaries that failed the validation to a file so users can troubleshoot

Reply:
Does logging the dictionary compromise any customer data? While having the failing report is good to log, the dictionary of data that generated could have sensitive info inside it and that could be undesirable to log out. Asking as not sure. Thanks for adjusting to more accurately determine failures! This will make testibility better in the long term.